src/plugins/data/common/es_query/es_query/types.ts

```diff
@@ -6,9 +6,29 @@
  * Side Public License, v 1.
  */
 
-import { IFieldType } from '../../index_patterns';
+import type { estypes } from '@elastic/elasticsearch';
+
+export interface IFieldSubType {
+  multi?: { parent: string };
+  nested?: { path: string };
+}
+export interface FieldBase {
+  name: string;
+  type: string;
+  subType?: IFieldSubType;
```

Comment:
Do you happen to know how subType is used? I'm sure I've looked it up before and forgotten. Would be good to document.